File: USACO/2021/JAN BRONZE/photos.cpp
using namespace std;
#include<iostream>
#include<vector>

int main() {
    int even = 0;
    int odd = 0;

	int n; cin >> n;

	for(int i = 0; i < n; i++) {
		int cow; cin >> cow;
        if(cow % 2) odd++;
        else even++; 
	}

    while (even < odd) {
        even += 1;
        odd -= 2;
    }
	
    if (odd + 1 < even) even = odd + 1;

	cout << odd + even << '\n';

    return 0;
}

File: USACO/2021/JAN BRONZE/photos.py
import sys
input = sys.stdin.readline

n = int(input())
even, odd = 0, 0

cows = [*map(int, input().split())]

for c in cows:
    if c % 2 == 0: even += 1
    else: odd += 1

while odd > even:
    odd -= 2
    even += 1

if even > odd + 1: even = odd + 1
print(even + odd)